users/tests/test_models.py

```diff
@@ -116,6 +119,75 @@ def test_implicit_oidc_login_user_login_entry_creation(client, oidc_client, user
         assert UserLoginEntry.objects.count() == 0
 
 
+@pytest.mark.parametrize('scope', ['', 'profile', 'email', 'profile email'])
+def test_implicit_oidc_login_id_token_content(
+        client, oidc_client, user, scope):
+    client.force_login(user)
+
+    data = {
+        'response_type': 'id_token token',
+        'client_id': oidc_client.client_id,
+        'redirect_uri': oidc_client.redirect_uris,
+        'scope': ' '.join(['openid', scope]),
+        'nonce': '123nonce456',
+    }
+
+    url = reverse('oidc_provider:authorize')
+    response = client.get(url, data, REMOTE_ADDR='1.2.3.4')
+    assert response.status_code == 302
+    redirect_location = response.get('location')
+    parsed_loc = urlparse(redirect_location)
+    params = parse_qs(parsed_loc.fragment)
+
+    id_token = params['id_token'][0]
+    id_token_data = jwt.decode(id_token, verify=False)
+
+    expected_keys = {
+        'aud', 'sub', 'exp', 'iat', 'iss',  'nonce',
+        'at_hash', 'auth_time',
+    } | ({
```

Comment:
This expression is a bit "clever", but I'll let it pass. :)
Or maybe it's idiomatic, I don't know.

Reply:
Especially lines 150 and 152